Marlin/src/feature/runout.h

```diff
@@ -285,16 +315,21 @@ class FilamentSensorBase {
       static inline void block_completed(const block_t* const) {}
 
       static inline void run() {
-        const bool out = poll_runout_state(active_extruder);
-        if (!out) filament_present(active_extruder);
-        #ifdef FILAMENT_RUNOUT_SENSOR_DEBUG
-          static bool was_out = false;
-          if (out != was_out) {
-            was_out = out;
-            SERIAL_ECHOPGM("Filament ");
-            serialprintPGM(out ? PSTR("OUT\n") : PSTR("IN\n"));
-          }
+        #if ENABLED(FILAMENT_RUNOUT_SENSOR_DEBUG)
+          static bool was_out[NUM_RUNOUT_SENSORS] = { false };
         #endif
+        LOOP_L_N(s, NUM_RUNOUT_SENSORS) {
```

Comment:
Is this looking a all sensors even when `WATCH_ALL_RUNOUT_SENSORS` is not enabled?

Reply:
The state is checked for all sensors, however a runout is only triggered for the active extruder / sensor or when WATCH_ALL_RUNOUT_SENSORS is enabled.